### PR TITLE
HUD CE APR

### PR DESCRIPTION
**Pull request by eanders**

### CODE REVIEW

File: drivers/hud_apr/app/models/hud_apr/generators/ce_apr/fy2020/question_four.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    QUESTION_TABLE_NUMBERS = ['Q4a'].freeze
+
+    def needs_ce_assessments?
+      true
+    end
+
+    def run_question!
+      @report.start(QUESTION_NUMBER, QUESTION_TABLE_NUMBERS)
+
+      # FIXME: this needs to inspect the resulting AprClients to determine which projects to include
```
Comment (awisema) [inline drivers/hud_apr/app/models/hud_apr/generators/ce_apr/fy2020/question_four.rb]: ?

File: drivers/hud_apr/app/models/hud_apr/generators/ce_apr/fy2020/question_nine.rb [hunk 2]
```diff
@@ -0,0 +1,179 @@
+###
+# Copyright 2016 - 2021 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module HudApr::Generators::CeApr::Fy2020
+  class QuestionNine < HudApr::Generators::Shared::Fy2020::QuestionNine
```
Comment (awisema) [inline drivers/hud_apr/app/models/hud_apr/generators/ce_apr/fy2020/question_nine.rb:8]: This doesn't seem to really be related to `HudApr::Generators::Shared::Fy2020::QuestionNine` should it inherit from `Base` directly?

File: drivers/hud_apr/app/models/hud_apr/generators/ce_apr/fy2020/question_ten.rb [hunk 3]
```diff
@@ -0,0 +1,133 @@
+###
+# Copyright 2016 - 2021 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module HudApr::Generators::CeApr::Fy2020
+  class QuestionTen < HudApr::Generators::Shared::Fy2020::QuestionTen
```
Comment (awisema) [inline drivers/hud_apr/app/models/hud_apr/generators/ce_apr/fy2020/question_ten.rb:8]: This one too -- does it use any of the shared Q10?

File: drivers/hud_apr/app/controllers/hud_apr/ce_apr/apr_concern.rb [hunk 4]
```diff
@@ -0,0 +1,39 @@
+###
+# Copyright 2016 - 2021 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module HudApr::Apr::AprConcern
```
Comment (awisema) [inline drivers/hud_apr/app/controllers/hud_apr/ce_apr/apr_concern.rb]: Is this actually used?

**Review by awisema — CHANGES_REQUESTED** [review]

File: drivers/hud_apr/app/models/hud_apr/generators/ce_apr/fy2020/question_four.rb (hunk 1 above)
Comment (eanders) [reply drivers/hud_apr/app/models/hud_apr/generators/ce_apr/fy2020/question_four.rb]: An initial comment that turned out to be unnecessary!

File: drivers/hud_apr/app/models/hud_apr/generators/ce_apr/fy2020/question_nine.rb (hunk 2 above)
Comment (eanders) [reply drivers/hud_apr/app/models/hud_apr/generators/ce_apr/fy2020/question_nine.rb:8]: I think you are correct.

File: drivers/hud_apr/app/models/hud_apr/generators/ce_apr/fy2020/question_ten.rb (hunk 3 above)
Comment (eanders) [reply drivers/hud_apr/app/models/hud_apr/generators/ce_apr/fy2020/question_ten.rb:8]: I believe so

File: drivers/hud_apr/app/controllers/hud_apr/ce_apr/apr_concern.rb (hunk 4 above)
Comment (eanders) [reply drivers/hud_apr/app/controllers/hud_apr/ce_apr/apr_concern.rb]: Nope, that was a bad copy paste cleanup job.

File: drivers/hud_apr/app/models/hud_apr/generators/ce_apr/fy2020/question_ten.rb (hunk 3 above)
Comment (eanders) [reply drivers/hud_apr/app/models/hud_apr/generators/ce_apr/fy2020/question_ten.rb:8]: hmm, tests fail miserably if it inherits from base.  I'm going to leave it, no harm in the current inheritance.

**Review by awisema — APPROVED** [review]
Thanks for doing this!

**Pull request merged**